beaconclient/util.go

```diff
@@ -30,28 +31,53 @@ func parseBroadcastModeString(s string) (BroadcastMode, bool) {
 	return b, ok
 }
 
-func fetchBeacon(method, url string, payload, dst any, timeout *time.Duration, headers http.Header) (code int, err error) {
+func makeJSONRequest(method, url string, payload any) (*http.Request, error) {
+	payloadBytes, err := json.Marshal(payload)
+	if err != nil {
+		return nil, fmt.Errorf("could not marshal request: %w", err)
+	}
+	req, err := http.NewRequest(method, url, bytes.NewReader(payloadBytes))
+	if err != nil {
+		return nil, fmt.Errorf("invalid request for %s: %w", url, err)
+	}
+	// Set content-type
+	req.Header.Add("Content-Type", "application/json")
+	return req, nil
+}
+
+func makeSSZRequest(method, url string, payload any) (*http.Request, error) {
+	if payloadBytes, ok := payload.([]byte); ok {
```

Comment:
the more typical pattern is checking for `!ok` and returning the error, instead of working in a nested way